### PR TITLE
handle more proto details

### PR DESCRIPTION
**Pull request by lthiery**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -18,6 +23,19 @@ pub use beacon::*;
 
 pub type Result<T = ()> = std::result::Result<T, Error>;
 
+pub enum Payload {
+    CellAttach(CellAttach),
+    CellScan(CellScan),
+    Beacon(Beacon),
+}
+
+pub struct Message {
```
Comment (vihu) [inline src/lib.rs:34]: Could you please `#[derive(debug)]` on some of these structs (I find it useful in general)

**Comment by vihu** [comment]
I like this in general, spotmapper-ingest wouldn't necessarily have to depend on helium-proto if it can be avoided.

I do think it would be good to expose sign for the `Message` which would basically sign the internal payload. Thoughts?

**Comment by lthiery** [comment]
> I do think it would be good to expose sign for the `Message` which would basically sign the internal payload. Thoughts?

Yes, I think that's the logical next step. Was just trying this first one out for size...

File: src/lib.rs (hunk 1 above)
Comment (lthiery) [reply src/lib.rs:34]: good point

**Comment by lthiery** [comment]
Re: Signing

There's actually quite a bit of extra work for signing the merits its own PR for review. So I'll land this and open that one up later today.

**Pull request merged**